src/adversarial/neurify.jl

```diff
@@ -0,0 +1,306 @@
+"""
+    Neurify(max_iter::Int64, tree_search::Symbol)
+
+Neurify combines symbolic reachability analysis with constraint refinement to minimize over-approximation of the reachable set.
+
+# Problem requirement
+1. Network: any depth, ReLU activation
+2. Input: AbstractPolytope
+3. Output: AbstractPolytope
+
+# Return
+`CounterExampleResult` or `ReachabilityResult`
+
+# Method
+Symbolic reachability analysis and iterative interval refinement (search).
+- `max_iter` default `10`.
+
+# Property
+Sound but not complete.
+
+# Reference
+[S. Wang, K. Pei, J. Whitehouse, J. Yang, and S. Jana,
+"Efficient Formal Safety Analysis of Neural Networks,"
+*CoRR*, vol. abs/1809.08098, 2018. arXiv: 1809.08098.](https://arxiv.org/pdf/1809.08098.pdf)
+[https://github.com/tcwangshiqi-columbia/Neurify](https://github.com/tcwangshiqi-columbia/Neurify)
+"""
+
+@with_kw struct Neurify
+    max_iter::Int64     = 100
+    tree_search::Symbol = :DFS # only :DFS/:BFS allowed? If so, we should assert this.
+end
+
+# Why can "interval" be an AbstractPolytope?
+struct SymbolicInterval{F<:AbstractPolytope}
```

Comment:
We spoke about star sets in the previous meeting; is this already the stand-in for a star set, or is the idea that eventually `F<:Star` but the rest remains the same? Since `interval` (which may be misnamed now, as already noted) is always the input set, I'm wondering whether a refactoring here makes sense as well.

Reply:
This is different from a star set. If we only have Low (or Up) and interval, then it is a star set. F will always be <:AbstractPolytope. I suggest changing the `interval` to `domain` to avoid confusion.